File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution(object):
    def waysToSplitArray(self, nums):
        total_sum = sum(nums)
        n=len(nums)
        currSum=0
        count=0
        for i in range(n-1):
            currSum+=nums[i]
            if(currSum >= total_sum - currSum):
                count+=1
        return count
        

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        
        long long total_sum = 0;
        int n = nums.size();
        for(auto& num : nums){
            total_sum += num;
        }
        int count=0;
        long long currSum=0;
        for(int i=0;i<n-1;i++){
            currSum+=nums[i];

            if(currSum >= total_sum - currSum){ 
                count++;
            }
        }
        return count;
    }
};